app/controllers/entities_controller.rb

```diff
@@ -31,7 +32,8 @@ def attach
   # Common discard handler for all core controllers.
   #----------------------------------------------------------------------------
   def discard
-    @attachment = params[:attachment].constantize.find(params[:attachment_id])
+    attachment = find_class(params[:attachment])
+    @attachment = attachment.find(params[:attachment_id])
     entity.discard!(@attachment)
```

Comment:
Maybe to leave one line find_class(params[:attachment]).find(params[:attachment_id]) ? here